.github/ISSUE_TEMPLATE/release.md

```diff
@@ -0,0 +1,94 @@
+# Release checklist
+
+## Step-by-step release process
+
+### Prepare release PR
+  - [ ] Create a branch named `release-{version}`:
+    - [ ] If it's a patch release, based the new branch out of the one that
+      contains the fixes. For example, if we added some fixes to the already
+      released `release-0.12` branch, then it will be the base for our `0.12.1`
+      release.
+  - [ ] Update `version` and `appVersion` in `deploy/helm-chart/Chart.yml`
+  - [ ] Update Promscale in `pkg/version/version.go` update. If required, update
+    the values for PrevReleaseVersion, PgVersionNumRange and
+    ExtVersionRangeString.
+    - [ ] Update `EXTENSION_VERSION` if there is a change in ExtVersionRangeString
+  - [ ] If the release includes changes to prom-migrator, update the `version`
+    variable at `migration-tool/cmd/prom-migrator/main.go`.
+  - [ ] (optionally) Update `prom-migrator` version in `.goreleaser.yml` (2 places)
+  - [ ] Run `./scripts/generate-deploy-script.sh`
+  - [ ] Finalize CHANGELOG, adding release version and date
+  - [ ] Commit: `git commit -a -m "Prepare for the 0.1.0-alpha.4 release"`
+  - Create PR:
+    - [ ] For major/minor releases the PR should target the master branch.
+    - [ ] For patch releases the PR should target the release branch it was
+      based of.
+  - Wait for review and merge.
+
+### Prepare git tag
+  - [ ] Pull the branch were the PR was merged, it should be `master` for
+    major/minor and `release-{version}` for patch releases. Triple check that
+    you have the right commit checked out.
+  - [ ] tag commit: `git tag -a 0.1.0-alpha.1 -m "Release 0.1.0-alpha.1"`
+  - [ ] push tag to main repo (not in fork!) `git push origin 0.1.0-alpha.1`
+
+### Create GitHub release notes
+
+[ ] The `goreleaser` GitHub Action will automatically create a new draft release with the generated binaries, docker images, and a changelog attached. When it is created contact PM to validate if release notes are correct and click green publish button.
```

Comment:
I think we need to update this to add the requirements section of the Release notes